library/Versions.re

```diff
@@ -145,3 +146,15 @@ let getRemoteVersions = () => {
      )
   |> Lwt.return;
 };
+
+let throwIfInstalled = versionName => {
+  getInstalledVersions()
+  |> Result.fold(
+       _ => Lwt.return(),
```

Comment:
If we would fold it to an empty array, we could push down the knowledge of Lwt and take out one indentation level:

```reason
let installed = getInstalledVersions()
|> Result.fold(_ => [||], xs => xs) // would return an empty array for errors
|> Array.exists(x => Local.(x.name == versionName))

if (installed) {
  Lwt.fail(Already_installed(versionName));
} else {
  Lwt.return();
}
```

Reply:
that being said, I'm really not sure we need to even care about Lwt here. We can stick to `isVersionInstalled` like you mentioned before that, and leave the logic of throwing an error to the executable itself (which also handles the error):

```reason
// library/Versions.re

let isVersionInstalled = versionName =>
  getInstalledVersions()
  |> Result.fold(_ => [||], xs => xs) // would return an empty array for errors
  |> Array.exists(x => Local.(x.name == versionName));
```

```reason
// executable/Install.re

let%lwt _ = if (Versions.isVersionInstalled(someVersion)) {
  Lwt.fail(Already_installed(someVersion));
} else {
  Lwt.return();
}
```